packages/flutter/lib/src/rendering/semantics_event.dart

```diff
@@ -121,3 +121,31 @@ class ScrollCompletedSemanticsEvent extends SemanticsEvent {
     return map;
   }
 }
+
+/// An event for a semantic announcement.
+///
+/// This should be used for announcement that are not seamlessly announced by
+/// the system as a result of a UI state change.
+///
+/// For example a camera application can use this method to make accessibility
+/// announcements regarding objects in the viewfinder.
+class AnnounceSemanticsEvent extends SemanticsEvent {
+
+  AnnounceSemanticsEvent(this.message, this.textDirection) : super('announce');
+
+  /// The message to announce.
```

Comment:
this does not satisfy our style guide for documentation.

Please include more detail. For example, can this contain newlines? How long should it be? What are some good practices and bad practices?

Reply:
I'm not sure about newlines, right now it's only used for TTS so it doesn't really matter, but in the future if we display this thing in some way we may want to allow/dis-allow newlines...

I don't think we have any guidelines for the length of the message... If your application needs to announce something long.. go for it...

The class doc comment talks about practices for this event, not sure if we have any practice recommendations for the specific text string...